Physical_Engine/Query/MaterialComposition.cs

```diff
@@ -63,25 +63,50 @@ public static MaterialComposition MaterialComposition(this ISurface surface)
         {
             if (surface.Construction == null)
             {
-                Engine.Reflection.Compute.RecordError("The ISurface MaterialComposition could not be calculated as no IConstruction has been assigned.");
+                Engine.Reflection.Compute.RecordError("The ISurface MaterialComposition could not be queried as no IConstruction has been assigned.");
                 return null;
             }
             return surface.Construction.IMaterialComposition();
         }
 
         /***************************************************/
 
-        [Description("Gets all the Materials a BulkSolid is composed of and in which ratios")]
-        [Input("bulkSolids", "The BulkSolids to get the MaterialComposition from")]
-        [Output("materialComposition", "The kind of matter the BulkSolids is composed of and in which ratios", typeof(Ratio))]
-        public static MaterialComposition MaterialComposition(this BulkSolids bulkSolids)
+        [Description("Gets all the Materials a SolidBulk is composed of and in which ratios")]
+        [Input("solidBulk", "The SolidBulk to get the MaterialComposition from")]
+        [Output("materialComposition", "The kind of matter the SolidBulk is composed of and in which ratios", typeof(Ratio))]
+        public static MaterialComposition MaterialComposition(this SolidBulk solidBulk)
         {
-            if (bulkSolids.MaterialComposition == null)
+            if (solidBulk == null)
             {
-                Engine.Reflection.Compute.RecordError("The BulkSolids MaterialComposition could not be calculated as no Materials have been assigned.");
                 return null;
             }
-            return Matter.Create.MaterialComposition(bulkSolids.MaterialComposition.Materials, bulkSolids.MaterialComposition.Ratios);
+
+            if (solidBulk.MaterialComposition == null)
+            {
+                Engine.Reflection.Compute.RecordError("The SolidBulk MaterialComposition could not be queried as no Materials have been assigned.");
+                return null;
+            }
+            return Matter.Create.MaterialComposition(solidBulk.MaterialComposition.Materials, solidBulk.MaterialComposition.Ratios);
```

Comment:
couldn't you simply return `solidBulk.MaterialComposition` here?

Reply:
yes, you can!